crates/reth/evm/src/precompiles/schnorr.rs

```diff
@@ -0,0 +1,65 @@
+use std::array::TryFromSliceError;
+
+use revm::primitives::{PrecompileError, PrecompileOutput, PrecompileResult};
+use revm_primitives::Bytes;
+use strata_crypto::verify_schnorr_sig;
+use strata_primitives::buf::{Buf32, Buf64};
+use thiserror::Error;
+
+/// 32 Bytes: Public key
+/// 32 Bytes: Message hash
+/// 64 Bytes: Schnorr Signature
+struct SchnorrInput {
+    public_key: Buf32,
+    message_hash: Buf32,
+    signature: Buf64,
+}
+
+#[derive(Debug, Error)]
+enum SchnorrError {
+    #[error("Input not exactly 128 bytes")]
+    InvalidInput,
+
+    #[error("Slice conversion failed: {0}")]
+    FixedBytes(#[from] TryFromSliceError),
+}
+
+fn parse_schnorr_input(input: &Bytes) -> Result<SchnorrInput, SchnorrError> {
+    // validate if the length is 128 bytes or not
+    if input.len() != 128 {
+        return Err(SchnorrError::InvalidInput);
+    }
```

Comment:
So reading the [Reth impl of ecrecover](https://github.com/bluealloy/revm/blob/6f3d9f89d538896498274e1fb44ffb42f189922a/crates/precompile/src/secp256k1.rs#L16), which is the closest equivalent, it seems like the convention in Ethereum is to actually pad to the length needed (also 128 bytes in their case) and just ignore bytes past the end.

That's not the most narrowly correct solution in my opinion, but it does means that we'll make it so we never have to reject an input without *trying* to verify it.  So we're "defining away" invalid input by defining a sensible behavior for it.

Reply:
[Disagree with printing a warning](https://github.com/alpenlabs/strata/pull/583#discussion_r1904375540, this isn't something we can control and it's not what's done in the ecrecover design.